src/Microsoft.Diagnostics.DebugServices/IMemoryRegionService.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.Diagnostics.DebugServices
+{
+    /// <summary>
+    /// Enumerate virtual address regions, their protections, and usage.
+    /// </summary>
+    public interface IMemoryRegionService
+    {
+        IEnumerable<IMemoryRegion> EnumerateRegions();
+    }
+
+    /// <summary>
+    /// Represents a single virtual address region in the target process.
+    /// </summary>
+    public interface IMemoryRegion
```

Comment:
Style nit: could you put IMemoryRegion in it's own file? It is ok to put the enum's it references in the file with it but at the beginning.  I generally like flags and enums to be first in a source file.